tests/datasets/test_geo.py

```diff
@@ -375,20 +376,31 @@ def test_no_all_bands(self) -> None:
         with pytest.raises(AssertionError, match=msg):
             CustomSentinelDataset(root, bands=bands, transforms=transforms, cache=cache)
 
+    def test_single_res(self) -> None:
+        root = os.path.join('tests', 'data', 'raster', 'res_2-2_epsg_32631')
+        ds = RasterDataset(root, res=10.0)
+        assert ds.res == (10.0, 10.0)
+
+    def test_change_res(self) -> None:
+        root = os.path.join('tests', 'data', 'raster', 'res_2-2_epsg_32631')
+        ds = RasterDataset(root, res=10.0)
+        assert ds.res == (10.0, 10.0)
+        ds.res = 20.0  # type: ignore[assignment]
```

Comment:
As far as I can tell, mypy doesn't like it when your getter returns a single type but your setter accepts a union of types, maybe discussed here https://github.com/python/mypy/issues/3004